pkg/clustermesh/common/remote_cluster.go

```diff
@@ -256,6 +287,10 @@ func (rc *remoteCluster) watchdog(ctx context.Context, backend kvstore.BackendOp
 		rc.metricLastFailureTimestamp.SetToCurrentTime()
 		rc.metricTotalFailures.Set(float64(rc.failures))
 		rc.metricReadinessStatus.Set(metrics.BoolToFloat64(rc.isReadyLocked()))
+		if rc.cacheTTL > 0 && rc.cacheTTLTimer == nil {
+			rc.logger.Warn("Starting cache TTL timer", "configured TTL", rc.cacheTTL)
+			rc.cacheTTLTimer = time.NewTimer(rc.cacheTTL)
```

Comment:
IIUC since go 1.23, we can just use time.After:
```suggestion
			rc.cacheTTLTimer = time.After(rc.cacheTTL)
```

(+ fixing cacheTTLTimer type)

Also I am not sure we should enable this timer only only after the first failure. It looks more straightforward/less surprising to do this on the first connection. So I am thinking that this could be init-ed directly in `restartRemoteConnection` basically ~.